domain/snyk/codeaction.go

```diff
@@ -16,27 +16,83 @@
 
 package snyk
 
+import (
+	"errors"
+
+	"github.com/google/uuid"
+)
+
+// CodeAction represents a code action that can be executed by the client using an in-document menu.
+// This type should be created by the NewCodeAction or NewDeferredCodeAction functions.
+//
+// There are 3 types of code actions:
+// - No Edit + No Command - Deferred code action, which means that either DeferredEdit or DeferredCommand must be set.
+// - Only edit/Only command - Resolved immediately to run the edit/command.
+// - Both edit and command - Resolved immediately to run edit first and then command.
 type CodeAction struct {
-	/**
-	 * A short, human-readable, title for this code action.
-	 */
+	// Title is a short, human-readable, title for this code action.
 	Title string
 
-	/**
-	 * Marks this as a preferred action. Preferred actions are used by the `auto fix` command and can be targeted
-	 * by keybindings.
-	 *
-	 * A quick fix should be marked preferred if it properly addresses the underlying error.
-	 * A refactoring should be marked preferred if it is the most reasonable choice of actions to take.
-	 *
-	 * @since 3.15.0
-	 */
-	IsPreferred bool
-
-	/**
-	 * The workspace edit this code action performs.
-	 */
-	Edit WorkspaceEdit
-
-	Command Command
+	IsPreferred *bool
```

Comment:
Why the pointers again? Sorry to complain so much, but I don't think it's necessary :(

Reply:
Nullability